### ISSUE TITLE
Jsonpath Extractor

### ISSUE THREAD

**Issue by potter420** [pola-rs/polars#994]
#### Jsonpath Extractor for certain columns

Can we implement some jsonpath extractors to json text column (UTF8). I can think of 2 functions:
Series.extract_json_path_single -> return a single UTF8 columns, null or raise error up on encountering non-json values maybe?
Series.extract_json_path_all -> return series of list type, which then can be exploded to the original dataframe.

I'm thinking of using jsonpath_lib (https://github.com/freestrings/jsonpath)

What do you think?

**Comment by ritchie46** [comment]
That seems useful! If you can add it under a feature flag, I am all for.

**Comment by potter420** [comment]
I'm currently testing `extract_json_path_single `, put it under extract_jsonpath feature flag. But my implementation is kind of weird. I try to find a way to construct new array with null value without using the builder, this is what I come up with:
```rust
// Function to extract jsonpath as string
#[cfg(feature = "extract_jsonpath")]
fn extract_json<'a>(expr: &'a Compiled, json_str: &'a str) -> Option<Cow<'a, str>>{
    let json: Value = from_str(json_str).unwrap();
    let result = expr.select(&json).unwrap();
    if result.is_empty() {
        None
    } else {
        if result[0].is_string() {
            Some(Cow::Owned(String::from(result[0].as_str().unwrap())))
         } else if result[0].is_i64(){
            Some(Cow::Owned(result[0].as_i64().unwrap().to_string()))
         } else if result[0].is_f64() {
            Some(Cow::Owned(result[0].as_f64().unwrap().to_string()))
         } else if result[0].is_boolean() {
            Some(Cow::Owned(result[0].as_bool().unwrap().to_string()))
         } else {
            None
        }
    }
}
impl Utf8Chunked {
...
 #[cfg(feature = "extract_jsonpath")]
    pub fn extract_json_path_single(&self, json_path: &str) -> Result<Utf8Chunked> {
        let pat = Compiled::compile(json_path).unwrap();
        // Seem like scan are the only function that will let me put Option<> in it
        let mut ca: Utf8Chunked = self.into_iter().scan(None, |previous, opt_v| match opt_v {
            Some(value) => {
                *previous = Some(value);
                Some(extract_json(&pat, value))
            }
            None => {
                Some(None)
            }
        }).collect();
        ca.rename(self.name());
        Ok(ca)
    }
}
```

**Comment by ritchie46** [comment]
You can use [apply](https://docs.rs/polars/0.14.7/polars/chunked_array/ops/trait.ChunkApply.html#impl-ChunkApply%3C%27a%2C%20%26%27a%20str%2C%20Cow%3C%27a%2C%20str%3E%3E).

Something like:

```rust
 #[cfg(feature = "extract_jsonpath")]
    pub fn extract_json_path_single(&self, json_path: &str) -> Result<Utf8Chunked? {

        let pat = Compiled::compile(json_path)?;

        Ok(self.apply(|str_val| extract_json(&path, str_val)))  
    }
}


```

**Comment by potter420** [comment]
Thanks, I actually went with `apply_on_opt` so that it allow None to be return when encounter valid JSON, JSONPath but no match is found.

**Comment by ritchie46** [comment]
merged in #1033

**Comment by Yevgnen** [comment]
Is `extract_json_path_all` the same as `s.map(json.loads)` in pandas? How can I turn a column of string into a column of JSON objects(list, dict)?
